### ISSUE TITLE
images.txt - Url with slug is only the slug in text mode

### ISSUE THREAD

**Issue by Hotwire75** [johanneszab/TumblThree#227]
Hi, 

When downloading image metas with metadata format set to text mode, "Url with slug:" only shows the slug.  If set as json, the correct info does show up in "url-with-slug".

I know this was mostly working correctly in at least 1.0.8.36, although IIRC, I /think/ it was broken for hidden blogs then, too... (I don't bother with metas on the one hidden blog I get, comment on each is identical)

Thanks

**Comment by Hotwire75** [comment]
Digging around cluelessly(not a programmer lol), in TumblThree.Applications/Parser/TumblrApiJsonToText.cs there are several 
"string.Format(CultureInfo.CurrentCulture, Resources.UrlWithSlug, post.slug)" lines.

Would this possibly supposed to be "post.url_with_slug"?

**Comment by johanneszab** [comment]
Hi,

sorry for the late reply. Yep, you are right. All those strings should be _post.url_with_slug_ instead of _post.slug_. You can see the data model that is being parsed in this file: [TumblrApiJson.cs](https://github.com/johanneszab/TumblThree/blob/master/src/TumblThree/TumblThree.Applications/DataModels/TumblrApiJson.cs#L148).

Thanks for creating this issue. I'll update the binaries at some point this weekend.

**Comment by Taranchuk** [comment]
Also in the metadata in the Photo Url there are duplicate links if the post is a photoset. Example:

`Photo Url: https://78.media.tumblr.com/cdbcc23a533b9b063ccd77e247e78912/tumblr_n5w4e8tBX91rdfxn4o1_1280.jpg https://78.media.tumblr.com/cdbcc23a533b9b063ccd77e247e78912/tumblr_n5w4e8tBX91rdfxn4o1_1280.jpg https://78.media.tumblr.com/90b432acf308f0306aca014b7032e99a/tumblr_n5w4e8tBX91rdfxn4o2_1280.jpg https://78.media.tumblr.com/e439bd70dbdd75616807f26404eefaa3/tumblr_n5w4e8tBX91rdfxn4o3_1280.jpg https://78.media.tumblr.com/e807863d09dd2ea6b577dd5a9acef245/tumblr_n5w4e8tBX91rdfxn4o4_1280.jpg https://78.media.tumblr.com/391ddd7090b621cedd337503aead61fc/tumblr_n5w4e8tBX91rdfxn4o5_1280.jpg`

**Comment by johanneszab** [comment]
Well, I've changed a bit more as the data in the svc data is different compared to the api data. Sometimes there is the slug added to the url, sometimes not, even though it's actually named url_with_slug. Sometimes there seems to be no slug at all. There is also a `url` field which contains the slug in the api, but not in the svc data ...

I think that I had already noticed that issue when I was implementing the text post parsers more than two years ago, but I wasn't really sure what to do.
I've added now a separate slug field, which sometimes might be empty though ...

> Also in the metadata in the Photo Url there are duplicate links if the post is a photoset. Example:

Actually that was on purpose. The first photo url is the cover photo from the photoset. I've put the photoset urls now in a separate row, similar to the slug. I'm not sure if that's actually better. Sure, one could check if the post is a photoset, then not add the photo cover url. But the way it's not, the `Photo url` is always just a single url, and `Photo set url` might be empty or it contains all the urls from the photo set.

Well, I don't now. I personally don't use this data at all. So maybe it's the best if someone how actually uses the data implements this properly which reasonable fields.
